src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -17381,6 +17460,79 @@ public static void Main()
         }
 
         [Fact]
+        public void OverriddenPropertyWithDifferentTupleNamesInReturn()
```

Comment:
One test I'm not seeing is bad metadata.  For example what happens when I have:

C# 

``` csharp
public class C1
{
  public abstract void M((int a, int b) p);
}
```

IL (as decompiled into C#)

``` csharp 
public class C2 : C1
{
  public void M((int x, int y) p);
}
```

In the IL case C# isn't enforcing override hence it's possible for names to differ.  Which name should we be enforcing here?  Think we should have a test for this.